libs/langchain/langchain/document_loaders/pdf.py

```diff
@@ -71,22 +72,39 @@ def __init__(self, file_path: str):
         if "~" in self.file_path:
             self.file_path = os.path.expanduser(self.file_path)
 
-        # If the file is a web path, download it to a temporary file, and use that
+        # If the file is a web path or S3, download it to a temporary file, and use that
         if not os.path.isfile(self.file_path) and self._is_valid_url(self.file_path):
-            r = requests.get(self.file_path)
-
-            if r.status_code != 200:
-                raise ValueError(
-                    "Check the url of your file; returned status code %s"
-                    % r.status_code
-                )
-
-            self.web_path = self.file_path
             self.temp_dir = tempfile.TemporaryDirectory()
-            temp_pdf = Path(self.temp_dir.name) / "tmp.pdf"
-            with open(temp_pdf, mode="wb") as f:
-                f.write(r.content)
-            self.file_path = str(temp_pdf)
+            _, suffix = os.path.splitext(self.file_path)
+            temp_pdf = Path(self.temp_dir.name) / f"tmp{suffix}"
+            if self._is_s3_url(self.file_path):
+                try:
+                    import boto3
+                except ImportError:
+                    raise ImportError(
+                        "Could not import `boto3` python package. "
+                        "Please install it with `pip install boto3`."
+                    )
+                result = urlparse(self.file_path)
+                bucket = result.netloc
+                prefix = result.path.lstrip('/')
+                self.web_path = self.file_path
+                s3 = boto3.client("s3")
+                s3.download_file(bucket, prefix, temp_pdf)
+                self.file_path = str(temp_pdf)
```

Comment:
Should probably extract the s3 related code into a separate function `_load_s3_file`; can we reuse the s3 loader here?